pkg/controller/backup_controller.go

```diff
@@ -865,3 +877,134 @@ func encodeToJSONGzip(data interface{}, desc string) (*bytes.Buffer, []error) {
 
 	return buf, nil
 }
+
+// deleteVolumeSnapshot delete VolumeSnapshot created during backup, if its
+// This is used to avoid deleting namespace in cluster triggers the VolumeSnapshot deletion,
+// which will cause snapshot deletion on cloud provider, then backup cannot restore the PV.
+// If DeletionPolicy is Retain, just delete it. If DeletionPolicy is Delete, need to
+// change DeletionPolicy to Retain before deleting VS, then change DeletionPolicy back to Delete.
+func (c *backupController) deleteVolumeSnapshot(volumeSnapshots []*snapshotv1api.VolumeSnapshot,
```

Comment:
Generally could you add a few info/debug messages like `deleting VS xxxx`

Reply:
Modified.